File: src/Enterprise/C#/patterns/State.cs
namespace Genkidama.PatternExamples;
public static class StateExample { public static bool Run(){string T(string s,string a)=>s=="locked"&&a=="unlock"?"unlocked":s=="unlocked"&&a=="lock"?"locked":s;return T(T("locked","unlock"),"lock")=="locked";} }

File: src/Enterprise/VB.NET/patterns/State.vb
Imports System
Friend Module StateExample
    Friend Function Run() As Boolean
        Dim transition As Func(Of String,String,String)=Function(state,action)
            If state="locked" AndAlso action="unlock" Then Return "unlocked"
            If state="unlocked" AndAlso action="lock" Then Return "locked"
            Return state
        End Function
        Return transition(transition("locked","unlock"),"lock")="locked"
    End Function
End Module
